server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -659,20 +677,34 @@ public void shutdownNow() {
         stopCachedTimeThread();
         scheduler.shutdownNow();
         for (ExecutorHolder executor : executors.values()) {
-            if (executor.executor() instanceof ThreadPoolExecutor) {
-                executor.executor().shutdownNow();
+            ExecutorService es = executor.executor();
+            if (es instanceof ThreadPoolExecutor || es instanceof ForkJoinPool) {
+                es.shutdownNow();
             }
         }
     }
 
     public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
-        boolean result = scheduler.awaitTermination(timeout, unit);
+        long nanos = unit.toNanos(timeout);
```

Comment:
If I understand this correctly, this changes the behavior so that the provided timeout is the max time it will take to shut down all executors, whereas previously it could wait timeout time for each executor. Why did you change this behavior? How do you know this won't adversely impact things that might depend on the previous behavior?